### DISCUSSION TITLE
Markdown自定义布局

### DISCUSSION THREAD

**Discussion by yzhang921** [vuepress-theme-hope/vuepress-theme-hope#704] [Q&A] [answered]
https://vuepress-theme-hope.github.io/zh/guide/layout/custom/#%E5%B8%83%E5%B1%80%E6%8F%92%E6%A7%BD 

咨询一下markdown自定义布局，比如说想用sidebar-top在sidebar最上面插一点东西，这个slot要写在哪个文件

**Comment by Mister-Hope** [comment]
你用markdown 的 slot 是只在特定的页面生效，在主题中配置slot为你的自定义组件全局生效。两者互不影响都会插入。

**Reply by yzhang921** [reply]
还是推荐在主题配置中用组件的形式全局配置是吧，好有我继续测试发现我那个项目，markdown增强都没有生效 

```javascript
    mdEnhance: {
        enableAll: true,
        align: true,
        // 启用下角标功能
        sub: true,
        // 启用上角标
        sup: true,
        flowchart: true,
    },
```

**Reply by Mister-Hope** [reply]
全部的配置文件截图一下

**Reply by yzhang921** [reply]
回复真快，👍  
发现问题了，写错位置了，要写在themeConfig下面，我的锅😳

**Reply by Mister-Hope** [reply]
hhh，主要是你不是第一个这么错的人了....对我来说这属于常见错误配置

**Reply by yzhang921** [reply]
我还有一个需求不知道怎么实现比较合适，侧边栏的标题支持上角标，之前原生主题是自己集成主题做的，hope主题的继承总觉得有点不一样

**Reply by Mister-Hope** [reply]
https://github.com/vuepress-theme-hope/vuepress-theme-hope/issues/489#issuecomment-871057006

这个方式有什么问题么？我主题全程用的都是@theme别名，应该可以被继承。

**Reply by yzhang921** [reply]
其实是我一直没有搞懂，你说的这种继承具体怎么操作😓  
能做一个demo项目，演示一下吗

**Reply by Mister-Hope** [reply]
这个继承是由vuepress提供的啊，回头有空的时候我做一个吧。（有可能忘，所以我一两天没动静你要提醒我）

**Reply by yzhang921** [reply]
按照官方的继承操作，具体应该是这样的，和你说的不一样 
https://www.coder.work/article/7471717，操作方式是在.vuepress下新建个/theme文件夹在/index.js里面定义继承，然后按照约定增加想要覆盖的组件

![image](https://user-images.githubusercontent.com/6037522/123903757-e4694b80-d9a1-11eb-9036-dfa0e363f431.png)

你的操作是，在.vueprss/config.js里面， 所以就搞不懂了

**Reply by Mister-Hope** [reply]
> 按照官方的继承操作，具体应该是这样的，和你说的不一样 
> https://www.coder.work/article/7471717，操作方式是在.vuepress下新建个/theme文件夹在/index.js里面定义继承，然后按照约定增加想要覆盖的组件
> 
> ![image](https://user-images.githubusercontent.com/6037522/123903757-e4694b80-d9a1-11eb-9036-dfa0e363f431.png)
> 
> 你的操作是，在.vueprss/config.js里面， 所以就搞不懂了
> 

我的锅，我记错了，你把我的extends删掉，然后你自己在index extends

**Reply by Mister-Hope** [reply]
总之你必须显示指定要使用本地主题

**Comment by yzhang921** [comment]
> 我的锅，我记错了，你把我的extends删掉，然后你自己在index extends

😓  这我肯定试过嘛，结果是整个文档都不出来了

**Reply by Mister-Hope** [reply]
等我有空我看一下，别急，我下午开组会。理论物理硕士....

**Reply by Mister-Hope** [reply]
\[进度更新\]: 忙里偷闲试了一下，好像属实有点问题，我回头定位一下。

**Reply by Mister-Hope** [reply]
![image](https://user-images.githubusercontent.com/33315834/123914931-87c25c80-d9b2-11eb-8389-cfcf2b686dd1.png)
![image](https://user-images.githubusercontent.com/33315834/123914956-914bc480-d9b2-11eb-9713-a9af3f3a1721.png)

发生了什么？我是命中了什么规则么？

**Reply by Mister-Hope** [reply]
好像的确命中了规则....

```md
\[进度更新\]: 忙里偷闲试了一下，好像属实有点问题，我回头定位一下。
```

有效

**Comment by Mister-Hope** [comment, marked as answer]
很抱歉当时忘记跟进了这个问题，目前暂无维护V1主题的打算了，请参加指定讨论。V2是可以正常继承的，欢迎迁移到V2.
